specs/bep-v1.rst

```diff
@@ -62,468 +62,276 @@ another.
-     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
-    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-    |  Ver  |       Message ID      |      Type     |   Reserved  |C|
-    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-    |                            Length                             |
-    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-
-For BEP v1 the **Version** field is set to zero. Future versions with
-incompatible message formats will increment the Version field. A message
-with an unknown version is a protocol error and MUST result in the
-connection being terminated. A client supporting multiple versions MAY
-retry with a different protocol version upon disconnection.
-
-The **Message ID** is set to a unique value for each transmitted Request
-message. In Response messages it is set to the Message ID of the corresponding
-Request message. The uniqueness requirement implies that no more than 4096
-request messages may be outstanding at any given moment. For message types
-that do not have a corresponding response (Cluster Configuration, Index, etc.)
-the Message ID field is irrelevant and SHOULD be set to zero.
-
-The **Type** field indicates the type of data following the message header
-and is one of the integers defined below. A message of an unknown type
-is a protocol error and MUST result in the connection being terminated.
-
-The **Compression** bit "C" indicates the compression used for the message.
-
-For C=0:
-
--  The Length field contains the length, in bytes, of the uncompressed
-   message data.
-
--  The message is not compressed.
-
-For C=1:
-
--  The Length field contains the length, in bytes, of the compressed
-   message data plus a four byte uncompressed length field.
-
--  The compressed message data is preceded by a 32 bit field denoting
-   the length of the uncompressed message.
-
--  The message data is compressed using the LZ4 format and algorithm
-   described in http://www.lz4.org/.
-
-All data within the message (post decompression, if compression is in
-use) MUST be in XDR (RFC 1014) encoding. All fields shorter than 32 bits
-and all variable length data MUST be padded to a multiple of 32 bits.
-The actual data types in use by BEP, in XDR naming convention, are the
-following:
+Every message post authentication is made up of several parts:
+
+- A header length word
+- A **Header**
+- A message length word
+- A **Message**
+
+.. code-block:: none
+
+     0                   1
+     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |         Header Length         |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    /                               /
+    \            Header             \
+    /                               /
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |         Message Length        |
+    |           (32 bits)           |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
```

Comment:
Thats not what I mean. I don't like that we put number of bits in brackets for 32bit ints and not 16bit ints